### PR TITLE
image: Bump base image to ubuntu 22.04

### LINKED ISSUE #19864: docker: Runtime image build failed
Docker build for runtime-image is currently failed in master branch, please find below details.

There is a slight different between expected and actual binary version/output/location.

```
$  git rev-parse HEAD 
57ae1fbb939803048fa9cac4efb97b94dde9108b

$ make -C images runtime-image
...
 => [linux/amd64 rootfs  4/12] RUN --mount=type=bind,readwrite,target=/go/src/github.com/cilium/cilium/images/runtime     ./install-runtime-deps.sh                                                                                  10.0s
 => [linux/amd64 rootfs  5/12] COPY iptables-wrapper /usr/sbin/iptables-wrapper                                                                                                                                                       0.0s
 => [linux/amd64 rootfs  6/12] RUN --mount=type=bind,readwrite,target=/go/src/github.com/cilium/cilium/images/runtime     ./configure-iptables-wrapper.sh                                                                             0.1s
 => [linux/amd64 rootfs  7/12] COPY --from=llvm-dist /usr/local/bin/clang /usr/local/bin/llc /bin/                                                                                                                                    0.1s
 => [linux/amd64 rootfs  8/12] COPY --from=bpftool-dist /usr/local /usr/local                                                                                                                                                         0.0s
 => [linux/amd64 rootfs  9/12] COPY --from=iproute2-dist /usr/lib/libbpf* /usr/lib                                                                                                                                                    0.0s
 => [linux/amd64 rootfs 10/12] COPY --from=iproute2-dist /usr/local /usr/local                                                                                                                                                        0.1s
 => [linux/amd64 rootfs 11/12] COPY --from=gops-cni-builder /out/linux/amd64/bin/loopback /cni/loopback                                                                                                                               0.0s
 => [linux/amd64 rootfs 12/12] COPY --from=gops-cni-builder /out/linux/amd64/bin/gops /bin/gops                                                                                                                                       0.0s
 => [linux/amd64 test 2/8] COPY --from=rootfs / /                                                                                                                                                                                     0.5s
 => [linux/amd64 test 3/8] COPY --from=llvm-dist /test /test                                                                                                                                                                          0.0s
 => [linux/amd64 test 4/8] COPY --from=bpftool-dist /test /test                                                                                                                                                                       0.0s
 => [linux/amd64 test 5/8] COPY --from=iproute2-dist /test /test                                                                                                                                                                      0.0s
 => ERROR [linux/amd64 test 6/8] RUN /test/bin/cst -C /test/llvm                                                                                                                                                                      0.1s
------
 > [linux/amd64 test 6/8] RUN /test/bin/cst -C /test/llvm:
#0 0.087 === RUN: Command Test: llc command is in path
#0 0.087 --- FAIL
#0 0.087 duration: 528.872µs
#0 0.087 stdout: /usr/bin/llc
#0 0.087 
#0 0.087 Error: Expected string '/usr/local/bin/llc' not found in output '/usr/bin/llc
#0 0.087 '
#0 0.087 === RUN: Command Test: llc --version
#0 0.087 --- PASS
#0 0.087 duration: 2.343706ms
#0 0.087 stdout: LLVM (http://llvm.org/):
#0 0.087   LLVM version 10.0.0
#0 0.087   Optimized build.
#0 0.087   Default target: x86_64-unknown-linux-gnu
#0 0.087   Host CPU: znver2
#0 0.087 
#0 0.087   Registered Targets:
#0 0.087     bpf   - BPF (host endian)
#0 0.087     bpfeb - BPF (big endian)
#0 0.087     bpfel - BPF (little endian)
#0 0.087 
#0 0.087 === RUN: Command Test: clang command is in path
#0 0.087 --- FAIL
#0 0.087 duration: 423.855µs
#0 0.087 stdout: /usr/bin/clang
#0 0.087 
#0 0.087 Error: Expected string '/usr/local/bin/clang' not found in output '/usr/bin/clang
#0 0.087 '
#0 0.087 === RUN: Command Test: clang --version
#0 0.087 --- FAIL
#0 0.087 duration: 3.979585ms
#0 0.087 stdout: clang version 10.0.0 (https://github.com/llvm/llvm-project.git 425a44e84b18e222cc38ac40603db1653a80a6cb)
#0 0.087 Target: x86_64-unknown-linux-gnu
#0 0.087 Thread model: posix
#0 0.087 InstalledDir: /usr/bin
#0 0.087 
#0 0.087 Error: Expected string 'InstalledDir:\ /usr/local/bin' not found in output 'clang version 10.0.0 (https://github.com/llvm/llvm-project.git 425a44e84b18e222cc38ac40603db1653a80a6cb)
#0 0.087 Target: x86_64-unknown-linux-gnu
#0 0.087 Thread model: posix
#0 0.087 InstalledDir: /usr/bin
#0 0.087 '
#0 0.087 === RUN: Command Test: clang can compile a simple BPF program
#0 0.087 --- PASS
#0 0.087 duration: 7.911078ms
#0 0.087 
#0 0.087 ===========================
#0 0.087 ========= RESULTS =========
#0 0.087 ===========================
#0 0.087 Passes:      2
#0 0.087 Failures:    3
#0 0.087 Duration:    15.187096ms
#0 0.087 Total tests: 5
#0 0.087 
#0 0.087 FAIL
------
Dockerfile:63
--------------------
  61 |     COPY --from=bpftool-dist /test /test
  62 |     COPY --from=iproute2-dist /test /test
  63 | >>> RUN /test/bin/cst -C /test/llvm
  64 |     RUN /test/bin/cst -C /test/bpftool
  65 |     RUN /test/bin/cst -C /test/iproute2
--------------------
error: failed to solve: process "/bin/sh -c /test/bin/cst -C /test/llvm" did not complete successfully: exit code: 1
+ exit 1
make: *** [Makefile:34: runtime-image] Error 1
make: Leaving directory '/home/tammach/go/src/github.com/cilium/cilium/images'

```

### PR DESCRIPTION
**Pull request by sayboras**
### Description

The new images are coming from https://github.com/cilium/image-tools/pull/191

Also, I take this chance to fix the below issue

Fixes https://github.com/cilium/cilium/issues/19864

Signed-off-by: Tam Mach <tam.mach@cilium.io>

### Next step

- [ ] Upgrade our dev VMs to 22.04 if required. Tracked in https://github.com/cilium/packer-ci-build/pull/344

### CODE REVIEW

**Review by pchaigno — COMMENTED** [review]
> The new images are coming from https://github.com/cilium/image-tools/pull/191

How is that possible given that PR isn't merged yet?

**Comment by sayboras** [comment]
> How is that possible given that PR isn't merged yet?

In the linked PR, I added below note. Basically, I tried to avoid last time mistake, in which we merged image-tools PR without knowing if cillium is working, and blocks other work.

> Testing was done with temp commit, which was removed after.

**Review by michi-covalent — APPROVED** [review]

**Comment by pchaigno** [comment]
> In the linked PR, I added below note. Basically, I tried to avoid last time mistake, in which we merged image-tools PR without knowing if cillium is working, and blocks other work.
> 
> > Testing was done with temp commit, which was removed after.

Oh, I see. So the temp commit was actually a commit in the present PR not in the other one? Makes sense.

**Review by tklauser — APPROVED** [review]

**Comment by sayboras** [comment]
> EDIT: No actually, I still don't get it. The image referenced in this PR can't be the ones we'll have once we merge the other PR, so doesn't that mean we'll need to update this PR again?

This is the build from master after image-tool PR is merged, the image hash is the same as we are using git tree hash of directory content (but not git hash).

https://github.com/cilium/image-tools/runs/7882391413?check_suite_focus=true

**Review by aanm — APPROVED** [review]

File: images/builder/Dockerfile [hunk 1]
```diff
@@ -2,7 +2,7 @@
 # SPDX-License-Identifier: Apache-2.0
 
 ARG COMPILERS_IMAGE=quay.io/cilium/image-compilers:5569a29cea6b3ad50aeb03102aaf3dc03841197c@sha256:b15dbedb7c49816c74a765e2f6ecdb9359763b8e4e4328d794f48b9cefae9804
-ARG CILIUM_RUNTIME_IMAGE=quay.io/cilium/cilium-runtime:07d9b863089eea0adc70f97f16cbb6c72cf3f14f@sha256:93e56c8f7e7cf1e6103fe8e8b978a5ae057e501a7ac7b78b51abb3905ed6557d
```
Comment (pchaigno) [inline images/builder/Dockerfile]: I see that we are still using the old reference for this image in `.github/workflows/build-images-ci.yaml`. Is that expected?

Comment (sayboras) [reply images/builder/Dockerfile]: good catch, let me update the build-images-ci for the same. The step in this workflow is to build race detection image.

**Review by pchaigno — APPROVED** [review]
`test-verifier` will also need an update in the K8sVerifier manifest but that can be a follow up PR.

**Comment by sayboras** [comment]
/test-1.16-4.9

**Comment by sayboras** [comment]
Hit flake for test-1.16-4.9 with bpf verifier.

Previous run was successful, force push changes were only for Github action update. All reviews are in, so marking this PR ready to merge.

https://jenkins.cilium.io/job/Cilium-PR-K8s-1.16-kernel-4.9/2361/

**Pull request merged**